server/broadcast.go

```diff
@@ -227,6 +234,27 @@ func (bsm *BroadcastSessionsManager) refreshSessions() {
 	bsm.sel.Add(uniqueSessions)
 }
 
+func (bsm *BroadcastSessionsManager) pushSegInFlight(sess *BroadcastSession, seg *stream.HLSSegment) {
+	bsm.sessLock.Lock()
+	defer bsm.sessLock.Unlock()
+	sess.SegsInFlight = append(sess.SegsInFlight,
+		SegFlightMetadata{
+			startTime: time.Now(),
+			segDur:    time.Duration(seg.Duration * float64(time.Second)),
+		})
+}
+
+func (bsm *BroadcastSessionsManager) popSegInFlight(sess *BroadcastSession) {
+	bsm.sessLock.Lock()
+	defer bsm.sessLock.Unlock()
+	if len(sess.SegsInFlight) == 1 {
+		sess.SegsInFlight = nil
```

Comment:
Setting the slice explicitly to `nil` here because that would free the underlying array. https://yourbasic.org/golang/clear-slice/

Otherwise the underlying array for SegsInFlight will keep getting extended, and slice pointer moved, but will only get cleared after a stream gets over, and might lead to memory leak for extremely long streams.